<launch>

	<arg name="record_trajectory" default="false" />	
	<arg name="publish_vector_field" default="false" />
	
	<node pkg="path_planner" type="path_planner_node" name="path_planner_node" output="screen">
		<param name="frame_id" type="string" value="world" />
		<param name="surfFlag" type="int" value="-1" />
		<param name="surfToBeDef" type="int" value="1" />
		<param name="tangFlag" type="int" value="-1" />
		<param name="sigma_multiplier" type="double" value="10" />
		<param name="xyGain" type="double" value="0.1" />
		<param name="zGain" type="double" value="0.1" />
		<param name="Kgrad1" type="double" value="0.1" />
		<param name="Kgrad2" type="double" value="0.1" />
		<param name="Ktang" type="double" value="0.3" />
		<param name="dist_sensed_obs" type="double" value="0.8" />
		<param name="pcRaise" type="double" value="0.0" />
		<param name="safety_margin" type="double" value="0.35" />
		<param name="publish_vector_field" type="bool" value="$(arg publish_vector_field)"/>

		<remap from="map_in" to="octomap_full" />
		<remap from="pose_in" to="Robot_1/pose"/>
		<remap from="command_out" to="fcu/control"/>
	</node>

	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="hector_trajectory_server" output="screen">
		<param name="target_frame_name" type="string" value="world" />
		<param name="source_frame_name" type="string" value="Robot_1/base_link" />
	</node>

	<node pkg="rosbag" type="record" name="rosbag_record_trajectory"
	       args="-o $(find path_planner)/bags/trajectories/trajectory /octomap_full /occupied_cells_vis_array /path_planner_node/velocity_vector_marker /trajectory /tf"
	       if="$(arg record_trajectory)" />
</launch>
